x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java

```diff
@@ -146,7 +146,9 @@ public void testCloseUnassignedJobAndDatafeed() throws Exception {
         waitForDatafeed(jobId, numDocs1);
 
         // stop the only ML node
+        ensureGreen(); // replicas must be assigned, otherwise we could lose a whole index
```

Comment:
Just curious: is it only testing issue or can it happen in prod as well?

Reply:
Yes, it can happen in production as well.  If an index has only just been created, the primary exists and the replica doesn't yet, and you kill the node with the primary on then you lose the index.

The situation where the primary exists but not the replica shouldn't last very long, so I guess this isn't a problem in realistic cases.  In production you might be able to recover by restarting the node with the primary on to make the single copy of the index available again.  Or you might not care that much because you'd hardly indexed any data into it.

If this was a big risk then waiting for green before indexing any data into a user-created index would generally be done by the client program.  For internal indices the effect of this happening in production would be that you could lose the config for your very first ML job if the node holding the primary of the `.ml-config` index died immediately after creating that first job.  But at that time all they'd lose would be one job config.  It's not like this problem could cause someone who had been running ML for years to lose everything.  For that they'd have to lose 2 nodes simultaneously...

Reply:
Thanks for explanation, this was very helpful.